File: code/ruby/students/students.rb
require "csv"

def enter_values
  puts "Please enter a person: "
  person = Student.new
  person.name = gets

  puts "Please enter their dorm: "
  person.house = gets

  person.student_living_text(person)
end

class Student
  attr_accessor :name, :house
  def initialize
    @name = name;
    @house = house;
  end

  def student_living_csv
    CSV.open("students.csv", "wb") do |csv|
      csv << ["Student", "House"]
      csv << ["#{@name}", "#{@house}"]
    end
  end

  def student_living_text(person)
    File.open("students.txt" ,"a") do |file|
      file.puts("#{person.name} lived in #{person.house}")
    end
  end
end

enter_values
enter_values
enter_values

File: code/C/structs/students.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>

#include "structs.h"

#define STUDENTS 3

int main(void) {
  student students[STUDENTS];

  for (int i = 0; i < STUDENTS; i++) {
    printf("Please enter a person: ");
    students[i].name = GetString();
    printf("Please enter their dorm: ");
    students[i].dorm = GetString();
    printf("\n");
  }

  FILE* file = fopen("students.csv", "a");

  if (file != NULL) {
    for (int i = 0; i < STUDENTS; i++) {
      fprintf(file, "%s, %s\n", students[i].name, students[i].dorm);
    }
  }
}
